### PR TITLE
[Merged by Bors] - chore: warn on cache misses

### PR DESCRIPTION
**Pull request by Kha**
I only touched the parallel-curl branch, which I assume is the mostly widely used. Comparing toolchain files when using mathlib as a dependency would also be nice.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Cache/Requests.lean [hunk 1]
```diff
@@ -103,6 +103,8 @@ def downloadFiles (hashMap : IO.HashMap) (forceDownload : Bool) (parallel : Bool
           msg := msg ++ s!", {failed} failed"
         IO.eprintln msg
       IO.FS.removeFile IO.CURLCFG
+      if success + failed < done then
+        IO.eprintln "Warning: some files were not found in the cache. This usually means that your local checkout of mathlib4 has diverged from upstream."
```
Comment (Kha) [inline Cache/Requests.lean:107]: This may or may not be a mathematician-friendly message

Comment (semorrison) [inline Cache/Requests.lean:107]:
```suggestion
        IO.eprintln "Warning: some files were not found in the cache."
        IO.eprintln "This usually means that your local checkout of mathlib4 has diverged from upstream."
        IO.eprintln "If you push your commits to a branch of the mathlib4 repository, CI will build the oleans and they will be available later."
```

Comment (semorrison) [reply Cache/Requests.lean:107]: We could be really really explicit for the poor mathematicians. :-)

Either with the original message or this one, great!

bors d+

**Pull request closed without merging**